service/controller/chart/v1/resource/status/create.go

```diff
@@ -23,31 +23,28 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	// If a reason was added to the controller context something went wrong.
-	// So we set the CR status and return early.
-	if cc.Status.Reason != "" {
-		status := v1alpha1.ChartStatus{
-			Reason: cc.Status.Reason,
-			Release: v1alpha1.ChartStatusRelease{
-				Status: cc.Status.Release.Status,
-			},
-		}
-
-		err = r.setStatus(ctx, cr, status)
-		if err != nil {
-			return microerror.Mask(err)
-		}
-
-		return nil
-	}
-
 	releaseName := key.ReleaseName(cr)
 	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("getting status for release %#q", releaseName))
 
 	releaseContent, err := r.helmClient.GetReleaseContent(ctx, releaseName)
 	if helmclient.IsReleaseNotFound(err) {
 		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("release %#q not found", releaseName))
 
+		// If a reason was added to the controller context something went wrong.
```

Comment:
Can we explain more why this is done here and why the release should be checked first? I miss a bit of conceptual understanding of the surroundings.

Reply:
Yes good point I'll add more to the comments as its important.

The problem is the operator needs to do 2 things. Pull the chart tarballs and act as the agent talking to Helm. But Helm doesn't know anything about the tarballs. 

So if we hit a 404 for example we add that to the context in the release resource and then include it in the CR status. We plan to expose this via the UX so the user knows whats going on.

See https://github.com/giantswarm/chart-operator/blob/9ea78e193272868219fee6a8f6b59b950a701ece/service/controller/chart/v1/resource/release/create.go#L38-L63 for where its added to the context.